### PR TITLE
Feature/argodev 532

### PR DESCRIPTION
**Pull request by vidkris**
Implementing the Doxy calculations using the class structure

### CODE REVIEW

File: bgc_derivation/derivations/Derivation.m [hunk 1]
```diff
@@ -5,29 +5,36 @@
...
             %
             
             self.default_glob = '**/*.nc';
             
-            if nargin == 1
-                self.variables = variables;
-            else
-                self.variables = containers.Map;
-            end
+			switch nargin
```
Comment (MCazaly) [inline bgc_derivation/derivations/Derivation.m:26]: The indentation looks a bit off here.

File: bgc_derivation/derivations/Derivation.m [hunk 2]
```diff
@@ -39,6 +46,18 @@ function add_data(self, name, data)
             
             self.variables(name) = data;
         end
+
+		function add_data_coeff(self, name, data)
```
Comment (MCazaly) [inline bgc_derivation/derivations/Derivation.m:50]: Indentation looking off here too.

File: bgc_derivation/derivations/Derivation.m [hunk 3]
```diff
@@ -39,6 +46,18 @@ function add_data(self, name, data)
...
+            % Add a single coeff by name.
+            %
+            % derivation.add_add_data_coeff(name, data)
+            % where
+            %  name is the name of the supplied variable.
+            %  data is the underlying array for the supplied variable.
+            %
+            
+            [self.coeffs(:).name] =data;
```
Comment (MCazaly) [inline bgc_derivation/derivations/Derivation.m:59]: Space after `=`?

**Review by MCazaly — CHANGES_REQUESTED** [review]
This looking good! Some minor comments added.

File: bgc_derivation/derivations/Derivation.m (hunk 1 above)
Comment (vidkris) [reply bgc_derivation/derivations/Derivation.m]: done. very strange. It was not visible in MATLAB

File: bgc_derivation/derivations/Derivation.m (hunk 2 above)
Comment (vidkris) [reply bgc_derivation/derivations/Derivation.m:50]: done

File: bgc_derivation/derivations/oxygen/DOXY_201_202_202.m [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        function output = compute_parameter(self)
+            % Primary method for dissolved oxygen derivation.
+            %
+            % derivation.compute_parameter()
+            %
+            
+            % Fetch all required parameters from source data
+
+            bphasedoxy = self.variables('BPHASE_DOXY');
+			rphasedoxy = self.variables('RPHASE_DOXY');
```
[the remark this answers is not in the document]
Comment (vidkris) [reply bgc_derivation/derivations/oxygen/DOXY_201_202_202.m]: done.

File: bgc_derivation/derivations/oxygen/equations/doxy_202.m [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+%        temp: temperature measurement values from CTD
+%        psal: salinity measurement values from CTD
+%
+% This covers OPTODES AANDERAA_OPTODE_3830
+%
+% See https://archimer.ifremer.fr/doc/00287/39795/94062.pdf
+% equation 7.2.12 case 201_202_202
+%
+function doxy=doxy_202(bphase, rphase, pres, temp, psal,coeffs)
+    if ~exist('coeffs')
```
[the remark this answers is not in the document]
Comment (vidkris) [reply bgc_derivation/derivations/oxygen/equations/doxy_202.m]: Agree. have made the changes.

**Review by MCazaly — APPROVED** [review]
Looks good to me :sunglasses:

**Pull request merged**